Framework/Algorithms/src/SampleCorrections/CircularBeamProfile.cpp

```diff
@@ -44,7 +44,8 @@ CircularBeamProfile::CircularBeamProfile(const Geometry::ReferenceFrame &frame,
 IBeamProfile::Ray CircularBeamProfile::generatePoint(
     Kernel::PseudoRandomNumberGenerator &rng) const {
   V3D pt;
-  const double R = rng.nextValue() * m_radius;
+  const double rsq = m_radius * m_radius;
+  const double R = std::sqrt(rng.nextValue() * rsq);
   const double theta = rng.nextValue() * 360.0;
   pt.spherical(R, theta, m_beamIdx);
```

Comment:
The change around sampling the R value looks good. Just noticed something separate though - is it correct to pass m_beamIdx into the call to pt.spherical? Looks like the 3rd parameter is a double called phi whereas m_beamIdx is an integer (short integer to be precise)
If I'm right - next question is what should be passed in. If the beam was fixed along the z axis then I think this line should perhaps be pt.spherical(R, 90, theta) because we want random points spread across the xy plane. However we need to write this so it's generic and doesn't assume the beam is along z. You've probably seen the frame object used in this unit to tell us which direction is the beam. Perhaps worth having a chat later once you've checked out whether I am right about m_beamIdx